src/core/unif.ml

```diff
@@ -304,7 +307,7 @@ and solve_aux : term -> term -> problems -> unif_constrs = fun t1 t2 p ->
   | (Prod(a1,b1), Prod(a2,b2))
   | (Abst(a1,b1), Abst(a2,b2)) ->
      let (_,b1,b2) = Bindlib.unbind2 b1 b2 in
-     solve_aux a1 a2 {p with to_solve = (b1,b2) :: p.to_solve}
+     solve_aux ctx a1 a2 {p with to_solve = (ctx,b1,b2) :: p.to_solve}
 
   (* Other cases. *)
   | (Vari(x1)   , Vari(x2)   ) ->
```

Comment:
You should do like in eq_modulo here.

Reply:
You mean with the `when Bindlib.eq_vars`?

Reply:
yes, and checking for definitions of x1 and x2

Reply:
ah right